### ISSUE TITLE
Inconsistent `STORAGE_EMULATOR_HOST` format

### ISSUE THREAD

**Issue by and3rson** [fsspec/gcsfs#547]
`gcsfs` seems to treat `STORAGE_EMULATOR_HOST` as a URL, not as a host:port pair.

This is breaking the convention of other libraries (including official Google libs) that treat `STORAGE_EMULATOR_HOST` as a host, and thus makes it impossible to use `gcsfs` with other libraries, because:
- `gcsfs` expects `STORAGE_EMULATOR_HOST` to be `port://host:port`
- `gcloud.aio.storage` & `google.cloud.storage` expect `STORAGE_EMULATOR_HOST` to be `host:port`

STR:
```sh
$ STORAGE_EMULATOR_HOST=127.0.0.1:4443  # Running `fakegcs`
>>> import gcsfs
>>> fs = gcsfs.GCSFileSystem()
>>> fs.listdir('gs://fakebucket/foo/bar')
Traceback (most recent call last):
...
aiohttp.client_exceptions.InvalidURL: 127.0.0.1:4443/storage/v1/b/fakebucket/o?delimiter=/&prefix=foo/bar/
```

A backward-compatible fix would be to detect if the scheme prefix (`http://` or `https://`) is missing, and append it if it is. Probably something as simple as changing https://github.com/fsspec/gcsfs/blob/main/gcsfs/core.py#L117:

```py
_emulator_location = os.getenv("STORAGE_EMULATOR_HOST", None)
if not any(_emulator_location.startswith(scheme) for scheme in ('http://' ,'https://')):
    _emulator_location = f'http://{_emulator_location}'  # Default to http:// since most people use this locally with fake gcs server
```

**Comment by and3rson** [comment]
Fixed by #548.

**Comment by andunai** [comment]
Thanks!
